crates/optimism/cli/src/lib.rs

```diff
@@ -86,3 +93,90 @@ impl Cli {
         Self::try_parse_from(itr)
     }
 }
+
+impl<Ext: clap::Args + fmt::Debug> Cli<Ext> {
+    /// Execute the configured cli command.
+    ///
+    /// This accepts a closure that is used to launch the node via the
+    /// [`NodeCommand`](reth_cli_commands::node::NodeCommand).
+    ///
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use reth::cli::Cli;
+    /// use reth_node_optimism::OptimismNode;
+    ///
+    /// Cli::parse_args()
+    ///     .run(|builder, _| async move {
+    ///         let handle = builder.launch_node(OptimismNode::default()).await?;
+    ///
+    ///         handle.wait_for_node_exit().await
+    ///     })
+    ///     .unwrap();
+    /// ```
+    ///
+    /// # Example
+    ///
+    /// Parse additional CLI arguments for the node command and use it to configure the node.
+    ///
+    /// ```no_run
+    /// use clap::Parser;
+    /// use reth::cli::Cli;
+    ///
+    /// #[derive(Debug, Parser)]
+    /// pub struct MyArgs {
+    ///     pub enable: bool,
+    /// }
+    ///
+    /// Cli::parse()
+    ///     .run(|builder, my_args: MyArgs| async move {
+    ///         // launch the node
+    ///
+    ///         Ok(())
+    ///     })
+    ///     .unwrap();
+    /// ````
```

Comment:
maybe out of scope, but I would prefer these to be executable examples. easy to keep up to date with potential type name changes then using `cargo test --doc`.

Reply:
we can finally do this now actually

Reply:
either it's in scope and this pr fixes it, or open an issue pls